packages/contented-pipeline/src/Pipeline.ts

```diff
@@ -125,6 +126,56 @@ export abstract class ContentedPipeline {
     return path;
   }
 
+  /**
+   * Compute the file name without any linking.
+   * Do use this instead of extracting manually.
+   * @param rawFileName The raw file name from ParsedPath
+   */
+  protected computeFileName(rawFileName: string): string {
+    if (rawFileName.includes('#')) {
+      const splitNames = rawFileName.split('#');
+
+      /**
+       * We only want it if its exactly length of 2.
+       * Title#subtile is valid, but Title#subtitle#whatisthis should not be valid.
+       * If its not valid just use the name as it is.
+       */
+      if (splitNames.length === 2) {
+        return `${slugify(this.replacePrefix(splitNames[0]))}`;
+      }
+    }
+    return `${slugify(this.replacePrefix(rawFileName))}`;
+  }
+
+  /**
+   * Extract the internal link from the read file.
+   * Depending if extension of the files are provided,
+   * the link will be in ext or the file name
+   * @param parsedPath
+   */
+  protected computeFileInternalLink(parsedPath: ParsedPath): string {
```

Comment:
```suggestion
  protected computeFileFragment(parsedPath: ParsedPath): string {
```

Let's be pedantic since "InternalLink" might be more confusing.